src/wp-includes/template.php

```diff
@@ -699,16 +700,19 @@ function get_attachment_template() {
  * @return string The template filename if one is located.
  */
 function locate_template( $template_names, $load = false, $load_once = true, $args = array() ) {
+	$stylesheet_path = get_stylesheet_directory();
+	$template_path   = get_template_directory();
+
 	$located = '';
 	foreach ( (array) $template_names as $template_name ) {
 		if ( ! $template_name ) {
 			continue;
 		}
-		if ( file_exists( STYLESHEETPATH . '/' . $template_name ) ) {
-			$located = STYLESHEETPATH . '/' . $template_name;
+		if ( file_exists( $stylesheet_path . '/' . $template_name ) ) {
+			$located = $stylesheet_path . '/' . $template_name;
 			break;
-		} elseif ( is_child_theme() && file_exists( TEMPLATEPATH . '/' . $template_name ) ) {
-			$located = TEMPLATEPATH . '/' . $template_name;
+		} elseif ( is_child_theme() && file_exists( $template_path . '/' . $template_name ) ) {
```

Comment:
```suggestion
		} elseif ( $template_path !== $stylesheet_path && file_exists( $template_path . '/' . $template_name ) ) {
```

Reply:
Addressed in https://github.com/WordPress/wordpress-develop/pull/5160/commits/928936f22dcf2bb0c16e98aaa910ce65dfba1780